### DISCUSSION TITLE
Sonoff TRVZB Local Temperature Offset 2.5 degree limit

### DISCUSSION THREAD

**Discussion by arthurdent2095** [zigpy/zha-device-handlers#3025] [Q&A] [answered]
It looks like ZHA now supports more features for the TRVZB however local temperature offset is limited to +/- 2.5 degrees and the device is capable of +/- 7 degrees. 

Is this something I can override with a custom quirk (I might need a quick pointer to that) ? 

Thanks in advance

PS. The firmware for this device was recently updated on Z2M. I am not sure how ZHA gets firmware updates for OTA, if you happen to know that would be super useful too.

**Comment by TheJulianJES** [comment, marked as answer]
Did you check Home Assistant Core 2024.3.0 for the local temp offset limit?

**Comment by TheJulianJES** [comment]
As far as I saw, the update that was provided on the Z2M repo was reverted after it broke devices until they were factory-reset.

**Comment by arthurdent2095** [comment]
aaah coool .. did some more searching and found 

https://github.com/home-assistant/core/pull/109691

Looks like the range extension was already in hand (by you) and merged a week ago so as you say maybe its in 2024.3, I haven't had time to upgrade yet will try it this weekend. 

Yes re: the firmware I am following that, though not sure how ZHA gets new firmware for OTA

**Reply by TheJulianJES** [reply]
Only official OTA providers are used at the moment. Sonoff has one and control it themselves.
If they put up new firmware, you'd see it. (This might change a bit in the future)
